src/Clide/Solution/ReferenceNode.cs

```diff
@@ -43,5 +43,7 @@ public class ReferenceNode : ProjectItemNode, IReferenceNode
             this.AsReference()
                 ?.SourceProject
                 ?.AsProjectNode();
+
+        public string Path => this.AsReference()?.Path;
```

Comment:
This is doing the conversion every time the property is retrieved. I would use a lazy here, since the reference node pointed to by the reference node cannot change once constructed. Arguably, the `SourceProject` could be treated the same way. Thoughts @adalon ?